src/primaite/agents/git/git_agent.py

```diff
@@ -0,0 +1,249 @@
+                node_services[node['Name']].append([k for k, v in self.service_mapping.items() if v == 'TCP_SQL'][0])
+            if node['UDP'] != '-':
+                node_services[node['Name']].append([k for k, v in self.service_mapping.items() if v == 'UDP'][0])
+        
+        # Get stringified node map, service map, node services, observation and action history and current observation difference for prompt
+        node_str = "\n".join(f"{key}: {value}" for key, value in self.node_mapping.items()) # ID: NODE_NAME
+        services_str = '\n'.join(f'{i}: {service}' for i, service in enumerate(self.service_mapping))
+        node_services_str = "\n".join(f"{key}: {', '.join(str(val) for val in value) if value else 'NONE'}" for key, value in node_services.items())
+        obs_act_history_str = get_obs_act_history_str(self.env_history, env=self._env)
+        
+        if not obs_act_history_str:
+            obs_act_history_str = 'No observation history yet...'
+        obs_diff_str = obs_diff(EnvironmentState(env=self._env))
+        
+        # Build and return prompts
+        reasoning_prompt = LLM_REASONING_PROMPT.format(node_ids=node_str, services=services_str, node_services=node_services_str, obs_act_history=obs_act_history_str, current_obs_diff=obs_diff_str)
+        
+        # Format all except reasoning statement, which is added later (GIT forward pass) after it is generated
+        action_prompt = LLM_PROMPT.format(node_ids=node_str, services=services_str, node_services=node_services_str, obs_act_history=obs_act_history_str, current_obs_diff=obs_diff_str, reasoning_statement='{reasoning_statement}')
+        
+        return action_prompt, reasoning_prompt
+    
+    def get_node_action(self, text_output) -> AgentNodeAction:
+        """Parses the LLM action output into an AgentNodeAction
+        
+        Args:
+            text_output (str): LLM action output
+        
+        Returns:
+            AgentNodeAction: The parsed AgentNodeAction
+        """
+        
+        splits = text_output.split('.')
+        
+        # Check for no action
+        if splits[0] == '0':
+            return AgentNodeAction(node_name='NONE', node_property='NONE', property_action='NONE', service_name='NONE')
+        
+        node = str(self.node_mapping[splits[0]])
+        
+        match int(splits[1]):
+            case 1:
+                property_action = 'TURN_ON'
+                node_property = 'HARDWARE'
+            case 2:
+                property_action = 'TURN_OFF'
+                node_property = 'HARDWARE'
+            case 3:
+                property_action = 'RESET'
+                node_property = 'HARDWARE'
+            case 4:
+                property_action = 'PATCH'
+                node_property = 'SOFTWARE'
+            case 5:
+                property_action = 'PATCH'
+                node_property = 'SERVICE'
+
+        # It's a service patch, so look for another integer
+        if int(splits[1]) == 5:
```

Comment:
this could be part of the match case statement

Reply:
yes